### ISSUE TITLE
Nodes with a name that ends with a "$" failed to be stored in postgres

### ISSUE THREAD

**Issue by leungmanhin** [opencog/atomspace#721]
Currently a `(ConceptNode "$")` and a `(WordNode "$")` will be generated if we do e.g.
```
(nlp-parse "They Paid $1,000 Just to See This 5 Minutes Show")
```
when saving them to postgres will give errors like:
```
Can't perform query rc=-1 (7) ERROR: syntax error at or near "$";
Error while executing the query
	Query was: INSERT INTO Atoms (uuid, space, type, name, height, tv_type, stv_mean, stv_count, stv_confidence) VALUES (2033, 1, 7, $$$$$, 0, 5,            1,            1,            0);
Can't perform query rc=-1 (7) ERROR: duplicate key value violates unique constraint "spaces_pkey"
Key (space)=(1) already exists.;
Error while executing the query
	Query was: INSERT INTO Spaces (space, parent) VALUES (1, 1);
Can't perform query rc=-1 (7) ERROR: syntax error at or near "$";
Error while executing the query
	Query was: INSERT INTO Atoms (uuid, space, type, name, height, tv_type, stv_mean, stv_count, stv_confidence) VALUES (2033, 1, 7, $$$$$, 0, 5,            1,            1,            0);
Can't perform query rc=-1 (7) ERROR: syntax error at or near "$";
Error while executing the query
	Query was: INSERT INTO Atoms (uuid, space, type, name, height, tv_type, stv_mean, stv_count, stv_confidence) VALUES (856, 1, 101, $$$$$, 0, 1,            1,            0,            0);
Can't perform query rc=-1 (7) ERROR: syntax error at or near "$";
Error while executing the query
	Query was: INSERT INTO Atoms (uuid, space, type, name, height, tv_type, stv_mean, stv_count, stv_confidence) VALUES (856, 1, 101, $$$$$, 0, 1,            1,            0,            0);
```

This may corrupt the database and give a segfault next time when we try to load atoms from it...

**Comment by inflector** [comment]
I know what did this. I'll fix it.

**Comment by inflector** [comment]
Pull request #722 fixes this issue.

**Comment by inflector** [comment]
Fixed and #722 merged. 

@leungmanhin Please check this fixes your problem and close this issue as appropriate.

**Comment by leungmanhin** [comment]
Yes it is fixed. Thanks @inflector
